script.js

```diff
@@ -1,4 +1,200 @@
+  var diceNumber = randomInteger + 1;
+  console.log("Dice rolled result : " + diceNumber);
+  return diceNumber;
+};
+
+var rollDiceForPlayer = function () {
+  console.log("time to roll dice for player");
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRoll.push(rollDice());
+    counter += 1;
+  }
+  console.log(
+    "played 2 times for player's dice roll data : " + currentPlayerRoll
+  );
+  return (
+    "Hello player" +
+    currentPlayer +
+    ", <br><br> " +
+    "You rolled <br>" +
+    "Dice 1 : " +
+    currentPlayerRoll[0] +
+    "<br>" +
+    "Dice 2 : " +
+    currentPlayerRoll[1] +
+    "<br><br>" +
+    "Now, please choose which dice you want to place as your placement. Input 1 for dice 1. Input 2 for dice 2. "
+  );
+};
+
+var getPlayerScore = function (input) {
+  var playerScore;
+
+  // input == 1
+  if (input == 1) {
+    console.log("input == 1st dice");
+    var playerScore = Number(
+      String(currentPlayerRoll[0]) + String(currentPlayerRoll[1])
+    );
+    console.log(playerScore);
+    // store this player score to all player score
+    allPlayerScores.push(playerScore);
+    //need to clean up current player roll array
+    currentPlayerRoll = [];
+    return (
+      "okay, you choose 1st dice to be placed first. <br>" +
+      "Your result : " +
+      playerScore
+    );
+  }
+
+  // input == 2
+  if (input == 2) {
+    console.log("playerInput == 2nd dice");
+    var playerScore = Number(
+      String(currentPlayerRoll[1]) + String(currentPlayerRoll[0])
+    );
+    console.log(playerScore);
+    // store this player score to all player score
+    allPlayerScores.push(playerScore);
+    //need to clean up current player roll array
+    currentPlayerRoll = [];
+    return (
+      "okay, you choose 2nd dice to be placed first. <br>" +
+      "Your result : " +
+      playerScore
+    );
+  }
+
+  return (
+    "Player" + currentPlayer + ", your final choosen value is : " + playerScore
+  );
+};
+
+var getWinner = function () {
+  // winner 1
+  if (allPlayerScores[0] > allPlayerScores[1]) {
+    return "Player 1, you won";
+  }
+  // winner 2
+  if (allPlayerScores[1] > allPlayerScores[0]) {
+    return "Player 2, you won";
+  }
+  // tie
+  if (allPlayerScores[0] == allPlayerScores[1]) {
+    return "Player 1 and Player 2, you are tie!";
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log("check / game mode : " + gameCurrentMode);
+  console.log("current player : " + currentPlayer);
+
+  var myOutputValue = "";
+
+  if (gameCurrentMode == gameModeDiceRoll) {
+    console.log("current status : game mode is at DICE ROLL");
+    // let the player roll 2 times dice and display the message instruction
+    myOutputValue = rollDiceForPlayer();
+    //change game mode to choose dice
+    gameCurrentMode = gameModeChooseDice;
+    return myOutputValue;
+  }
+
+  if (gameCurrentMode == gameModeChooseDice) {
```

Comment:
Similar to the above comment, consider using else if statements here where appropriate rather than just if statements